server/src/test/java/org/elasticsearch/search/aggregations/metrics/ScriptedMetricIT.java

```diff
@@ -90,42 +91,57 @@
         protected Map<String, Function<Map<String, Object>, Object>> pluginScripts() {
             Map<String, Function<Map<String, Object>, Object>> scripts = new HashMap<>();
 
-            scripts.put("_agg['count'] = 1", vars ->
-                    aggScript(vars, agg -> ((Map<String, Object>) agg).put("count", 1)));
+            scripts.put("state['count'] = 1", vars ->
+                    aggScript(vars, state -> state.put("count", 1)));
 
-            scripts.put("_agg.add(1)", vars ->
-                    aggScript(vars, agg -> ((List) agg).add(1)));
+            scripts.put("state.list.add(1)", vars ->
```

Comment:
Some of these scripts and test cases had to be revised to add a `list` var under the agg state since `state` itself can't be custom initialized with an arbitrary type like `params._agg/_aggs` could.